Neos.ContentRepository.Core/Classes/Projection/ContentGraph/Filter/SearchTerm/SearchTermMatcher.php

```diff
@@ -16,41 +17,47 @@ class SearchTermMatcher
 {
     public static function matchesNode(Node $node, SearchTerm $searchTerm): bool
     {
-        return static::matchesSerializedPropertyValues($node->properties->serialized(), $searchTerm);
+        return static::matchesSerializedPropertyValues($node->properties->serialized(), $searchTerm, $node->aggregateId);
     }
 
-    public static function matchesSerializedPropertyValues(SerializedPropertyValues $serializedPropertyValues, SearchTerm $searchTerm): bool
+    public static function matchesSerializedPropertyValues(SerializedPropertyValues $serializedPropertyValues, SearchTerm $searchTerm, ?NodeAggregateId $nodeAggregateId = null): bool
     {
         if ($searchTerm->term === '') {
             return true;
         }
-        foreach ($serializedPropertyValues as $serializedPropertyValue) {
-            if (self::matchesValue($serializedPropertyValue->value, $searchTerm)) {
+        foreach ($serializedPropertyValues as $propertyName => $serializedPropertyValue) {
+            if (self::matchesValue($serializedPropertyValue->value, $searchTerm, $propertyName, $nodeAggregateId)) {
                 return true;
             }
         }
         return false;
     }
 
-    private static function matchesValue(mixed $value, SearchTerm $searchTerm): bool
+    private static function matchesValue(mixed $value, SearchTerm $searchTerm, string $propertyName, ?NodeAggregateId $nodeAggregateId): bool
     {
         if (is_array($value) || $value instanceof \ArrayObject) {
             foreach ($value as $subValue) {
-                if (self::matchesValue($subValue, $searchTerm)) {
+                if (self::matchesValue($subValue, $searchTerm, $propertyName, $nodeAggregateId)) {
                     return true;
                 }
             }
             return false;
         }
 
         return match (true) {
+            $value === null => false,
```

Comment:
i dont actually understand why its null? The `SerializedPropertyValue` ensures that its never null?
But maybe its the stupid `$subValue` iteration of mine? Good catch. I was trying to be clever and fully accurate ^^

Reply:
yep, it's definitively not the property value itself, it's a value within the property value, which is a complex VO collection in my case